pkg/client/fake/client.go

```diff
@@ -61,11 +66,33 @@ func NewFakeClientWithScheme(clientScheme *runtime.Scheme, initObjs ...runtime.O
 		}
 	}
 	return &fakeClient{
-		tracker: tracker,
+		tracker: versionedTracker{tracker},
 		scheme:  clientScheme,
 	}
 }
 
+func (t versionedTracker) Create(gvr schema.GroupVersionResource, obj runtime.Object, ns string) error {
+	if accessor, err := meta.Accessor(obj); err == nil {
```

Comment:
Wouldn't it make more sense to return the error here if its non-nil? Granted, that shouldn't happen in the first place, but if it does it would IMHO be nicer to return that instead of silently degrading